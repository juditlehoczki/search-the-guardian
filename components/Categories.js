import React from "react";
import { View, Text, TouchableOpacity } from "react-native";
import s from "../styles.js";

const Categories = (props) => {
  const { getArticlesBySection } = props;
  const categories = [
    "Culture",
    "Environment",
    "Film",
    "Food",
    "Music",
    "Science",
    "Technology",
    "Travel",
  ];
  return (
    <View>
      <Text style={s.sectionTitle}>
        Not sure what to read? Here are our favourite topics:
      </Text>
      {categories.map((category) => {
        return (
          <TouchableOpacity
            onPress={() => {
              getArticlesBySection(1, category.toLowerCase());
            }}
            key={category}
          >
            <View style={s.card}>
              <Text style={s.category}>{category.toUpperCase()}</Text>
              <Text style={s.arrow1}>&gt;</Text>
              <Text style={s.arrow2}>&gt;</Text>
            </View>
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

export default Categories;
