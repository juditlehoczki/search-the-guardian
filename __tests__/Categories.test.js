import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {TouchableWithoutFeedback} from 'react-native';

import Categories from '../components/Categories.js';

const fakeGetArticlesBySection = jest.fn();
const props = {
  getArticlesBySection: fakeGetArticlesBySection,
};

describe('Categories Component', () => {
  test('Snapshot - it renders correctly', () => {
    const snapshot = renderer.create(<Categories {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('When a category is pressed it calls getArticlesBySection function', () => {
    const fakeCategories = renderer.create(<Categories {...props} />);
    const categoryCards = fakeCategories.root.findAllByType(
      TouchableWithoutFeedback,
    );
    categoryCards[0].props.onPress();
    expect(fakeGetArticlesBySection).toHaveBeenCalled();
  });
});
