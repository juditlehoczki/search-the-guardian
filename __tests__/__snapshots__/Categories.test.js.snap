// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Component Snapshot - it renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "backgroundColor": "#54C1B7",
        "color": "white",
        "fontWeight": "bold",
        "marginVertical": 15,
        "paddingLeft": 15,
        "paddingVertical": 5,
      }
    }
  >
    Not sure what to read? Here are our favourite topics:
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "marginHorizontal": 5,
        "marginVertical": 3,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#0F406E",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 3,
          "padding": 10,
        }
      }
    >
      CULTURE
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#D1529E",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#54C1B7",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "marginHorizontal": 5,
        "marginVertical": 3,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#0F406E",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 3,
          "padding": 10,
        }
      }
    >
      ENVIRONMENT
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#D1529E",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#54C1B7",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "marginHorizontal": 5,
        "marginVertical": 3,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#0F406E",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 3,
          "padding": 10,
        }
      }
    >
      FILM
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#D1529E",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#54C1B7",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "marginHorizontal": 5,
        "marginVertical": 3,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#0F406E",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 3,
          "padding": 10,
        }
      }
    >
      FOOD
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#D1529E",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#54C1B7",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "marginHorizontal": 5,
        "marginVertical": 3,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#0F406E",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 3,
          "padding": 10,
        }
      }
    >
      MUSIC
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#D1529E",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#54C1B7",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "marginHorizontal": 5,
        "marginVertical": 3,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#0F406E",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 3,
          "padding": 10,
        }
      }
    >
      SCIENCE
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#D1529E",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#54C1B7",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "marginHorizontal": 5,
        "marginVertical": 3,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#0F406E",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 3,
          "padding": 10,
        }
      }
    >
      TECHNOLOGY
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#D1529E",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#54C1B7",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "marginHorizontal": 5,
        "marginVertical": 3,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#0F406E",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 3,
          "padding": 10,
        }
      }
    >
      TRAVEL
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#D1529E",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#54C1B7",
          "fontSize": 18,
        }
      }
    >
      &gt;
    </Text>
  </View>
</View>
`;
